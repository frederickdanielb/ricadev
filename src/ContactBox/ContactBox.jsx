import PropTypes from 'prop-types';
import { faEnvelope, faMapMarkedAlt, faPhoneAlt } from '@fortawesome/free-solid-svg-icons';
import { faFacebookF, faYoutube } from '@fortawesome/free-brands-svg-icons';
import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
const ContactBox = ({ className, contact }) => {
	return (
		<div className={'single-sidebar ' + className}>
			<h6 className="contact-title">Cont&aacute;ctanos</h6>
			{contact.address && (
				<p className="address">
					<FontAwesomeIcon icon={faMapMarkedAlt} />
					{contact.address}
				</p>
			)}
			{contact.phone && (
				<p>
					<FontAwesomeIcon icon={faPhoneAlt} /> {contact.phone}
				</p>
			)}
			{contact.email && (
				<a href={'mailto:' + contact.email}>
					<p>
						<FontAwesomeIcon icon={faEnvelope} /> {contact.email}
					</p>
				</a>
			)}
			<div className="social-box">
				<ul>
					{contact.social_networks.facebook && (
						<li>
							<a href={contact.social_networks.facebook}>
								<FontAwesomeIcon icon={faFacebookF} />
							</a>
						</li>
					)}
					{contact.social_networks.youtube && (
						<li>
							<a href={contact.social_networks.youtube}>
								<FontAwesomeIcon icon={faYoutube} />
							</a>
						</li>
					)}
				</ul>
			</div>
		</div>
	);
};

ContactBox.propTypes = {
	className: PropTypes.string,
	contact: PropTypes.object.isRequired,
};
export default ContactBox;
