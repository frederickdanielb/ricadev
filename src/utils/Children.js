import React from "react";

export function map(children, func, context) {
	let index = 0;
	return React.Children.map(children, (child) => {
		if (!React.isValidElement(child)) {
			return child;
		}
		const handle = func.call(context, child, index);
		index += 1;
		return handle;
	});
}
export function mapCloneElement(children, func, context) {
	return map(
		children,
		(child, index) =>
			React.cloneElement(child, {
				key: index,
				...func(child, index),
			}),
		context
	);
}
