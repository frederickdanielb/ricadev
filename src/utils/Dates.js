export const parseDate = (str) => {
	var ymd = str.split("-");
    
	return new Date(ymd[0], ymd[1] - 1, ymd[2]);
};

export const datediff = (first, second) => {
	// Take the difference between the dates and divide by milliseconds per day.
	// Round to nearest whole number to deal with DST.
	return Math.round((second - first) / (1000 * 60 * 60 * 24));
};

export const getDaysBetween = (firsDate, secondDate) =>
	datediff(parseDate(firsDate), parseDate(secondDate));
