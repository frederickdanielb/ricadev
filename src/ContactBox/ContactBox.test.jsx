import '@testing-library/jest-dom'
import { render, screen, userEvent } from './../../tests/utils/test-utils'
import { MockConfig } from "../../config/_mocks_/config";
import ContactBox from "./ContactBox";

describe("contact box component", () => {
	const propsWithData = MockConfig;
	let container = null;

	beforeEach(() => {
		container = render(<ContactBox config={propsWithData} />);
	});

	test("should render correctly", async () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
	test("should show params", () => {
		const email = screen.getByText(/test@test.cl/)
		expect(email).toBeInTheDocument()
		const address = screen.getByText(/Av. Los Leones 1071/)
		expect(address).toBeInTheDocument()
		const phone = screen.getByText(/562 2 476 5800/)
		expect(phone).toBeInTheDocument()
	})
	describe("and when component not have config", () => {
		beforeEach(() => {

			let propsWithoutData = {...MockConfig};
			propsWithoutData.contact.address = null;
			propsWithoutData.contact.phone = null;
			propsWithoutData.contact.email = null;
			propsWithoutData.contact.social_networks = {};
			container = render(<ContactBox config={propsWithoutData} />);
		});
		test("should not render params not mapped", () => {
			expect(container.asFragment()).toMatchSnapshot();
		});
	});
});
