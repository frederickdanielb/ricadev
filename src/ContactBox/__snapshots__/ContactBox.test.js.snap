// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact box component and when component not have config should not render params not mapped 1`] = `
<DocumentFragment>
  <div
    class="single-sidebar undefined"
  >
    <h6
      class="contact-title"
    >
      Contáctanos
    </h6>
    <div
      class="social-box"
    >
      <ul />
    </div>
  </div>
</DocumentFragment>
`;

exports[`contact box component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="single-sidebar undefined"
  >
    <h6
      class="contact-title"
    >
      Contáctanos
    </h6>
    <p
      class="address"
    >
      <i
        class="fas fa-map-marker-alt"
      />
      Av. Los Leones 1071
    </p>
    <p>
      <i
        class="fas fa-phone-alt"
      />
       +562 2 476 5800
    </p>
    <a
      href="mailto:test@test.cl"
    >
      <p>
        <i
          class="fas fa-envelope"
        />
        test@test.cl
      </p>
    </a>
    <div
      class="social-box"
    >
      <ul>
        <li>
          <a
            href="https://www.facebook.com/TurismoPanamericana/"
          >
            <i
              class="fab fa-facebook-f"
            />
          </a>
        </li>
        <li>
          <a
            href="https://www.youtube.com/channel/UCGiVnhb-GDcDujk1K8PC_8g"
          >
            <i
              class="fab fa-youtube"
            />
          </a>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
